sdk/src/main/java/io/opentelemetry/sdk/trace/SpanBuilderSdk.java

```diff
@@ -185,9 +183,10 @@ public Span startSpan() {
             samplingDecision.isSampled() ? TRACE_OPTIONS_SAMPLED : TRACE_OPTIONS_NOT_SAMPLED,
             tracestate);
 
+    if (!recordEvents) {
```

Comment:
This should be !recordEvents && !sampled

Reply:
Updated, though I'm not sure on cases when a span builder is sampled but not recording events. For example:
```java
Span span = tracer.spanBuilder("span").setRecordEvents(false).setSampler(Samplers.alwaysSample());
span.isRecordingEvents();  // returns true
```
Is this what we wanted?

Reply:
Chatted offline, all sampled spans will be recording events by default.

Reply:
So all sampled spans will record events, on top of these spans that people turn on record events manually will also record events.